drivers/counter/counter_mcux_ctimer.c

```diff
@@ -277,6 +279,32 @@ static int mcux_lpc_ctimer_init(const struct device *dev)
 	return 0;
 }
 
+static int mcux_lpc_ctimer_pm_action(const struct device *dev, enum pm_device_action action)
+{
+	switch (action) {
+	case PM_DEVICE_ACTION_RESUME:
+		break;
+	case PM_DEVICE_ACTION_SUSPEND:
+		break;
+	case PM_DEVICE_ACTION_TURN_OFF:
+		break;
+	case PM_DEVICE_ACTION_TURN_ON:
+		mcux_lpc_ctimer_init_common(dev);
+		break;
+	default:
+		return -ENOTSUP;
+	}
+	return 0;
+}
+
+static int mcux_lpc_ctimer_init(const struct device *dev)
+{
+	/* Rest of the init is done from the PM_DEVICE_TURN_ON action
+	 * which is invoked by pm_device_driver_init().
```

Comment:
you can't do this paradigm because PM device could (and most likely will) be off on most platforms and app configurations

Reply:
@decsny . This implementation was added per this comment.
https://github.com/zephyrproject-rtos/zephyr/pull/84945#discussion_r1957080778

Reply:
ok, I did not realize that having `CONFIG_PM_DEVICE=n` will still build all the PM device files